packages/semcom-core/docs/modules/specifications/pages/637440573-support-registering-components.adoc

```diff
@@ -0,0 +1,93 @@
+== Technical Specification: Support registering of components
+
+=== Author(s)
+
+* Stijn Taelemans
+
+=== References
+
+* Functional analysis: see semcom-core docs
+
+* Branch: `637440573-support-registering-components`
+* Projects:
+** https://github.com/digita-ai/semcom[semcom]
+
+=== Introduction
+
+==== Overview
+
+This specification is about part of the SDK that registers the Components on the web page using the `CustomElementRegistry`.
+
+=== Solution
+
+_This is a proposed solution and is open to change. If you feel some things can be done better, please discuss with this document's link:###authors[author(s)]._
+
+==== semcom-core
+
+===== Models
+
+====== ComponentMetadata
+
+Add the `tag: string` attribute to the existing `ComponentMetadata` model. We'll need it later for registering the component.
+
+Example: 
+
+[source, json]
+----
+{
+    "uri": "https://components.digita.ai/album",
+    "name": "Image Album Component",
+    "label":  "Component that renders images",
+    "author":  "digita.ai",
+    "tag": "dgt-album",
```

Comment:
How does a developer the preferred tag for a component? What happens if a developer configures a different preferred tag in the component's code compared to its metadata in the SemCom repository?

Reply:
I would actually not store the tag-name, or any metadata, in the component code. This info is/should not be needed internally, as it only describes the component to the outside world.